cpmonitor/views.py

```diff
@@ -506,6 +508,52 @@ def markdown_uploader_view(request):
     return HttpResponse(data, content_type="application/json")
 
 
+def mstr_view(request, municipality_key):
+    url = "https://www.marktstammdatenregister.de/MaStR/Einheit/EinheitJson/GetVerkleinerteOeffentlicheEinheitStromerzeugung"
+    params = {
+        "sort": "EinheitMeldeDatum-desc",
+        "page": 1,
+        "pageSize": 10000,
+        "filter": f"Energieträger~eq~'2495'~and~Gemeindeschlüssel~eq~'{municipality_key}'",
+    }
+    try:
+        r = requests.get(url, params)
+    except:
+        return HttpResponseServerError(500)
+
+    data = r.json()["Data"]
+
+    installed_by_year = dict()
+
+    START_YEAR = 2019
+    for entry in data:
+        if not entry["InbetriebnahmeDatum"]:
+            continue
+
+        m = re.match("/Date\((-?\d*)\)/", entry["InbetriebnahmeDatum"])
+        install_date = datetime.fromtimestamp(int(m.group(1)) / 1000)
+        year = START_YEAR if install_date.year < START_YEAR else install_date.year
```

Comment:
So the start year contains the Bruttoleistung from PVs from all the years before the start year? 
That was not really clear to me from the look of the graphic.

Reply:
Each bar in the bar diagram shows the accumulated power (Gesamtleistung) of all PV in the given city installed by the end of that year. It is not added power (Zubau). I talked to Philipp about it and he prefers added power (Zubau), so that might change, but for the time begin it is accumulated power.